source/utilities.cc

```diff
@@ -167,6 +167,57 @@ namespace aspect
     }
 
     template <int dim>
+    bool
+    in_2d_polygon(dealii::Point<2> &point,const std::vector<std::vector<double> > &pointList)
+    {
```

Comment:
What happens if you call this function with `dim==3`? Does it just ignore the 3rd component of each point in the list?